### PR TITLE
Clear logs after search

### LINKED ISSUE #4246: Clear the logs on a new search
as a logviewer user
with viewer permissions or higher
when i make a new search in the log viewer
then i expect the current results to disappear and have an indication that we are searching.

keeping the previous results is confusing since it isn't clear to me when the new search has been applied. We should clear the current log lines while we wait for the query to complete.

### PR DESCRIPTION
**Pull request by OfTheDelmer**
Closes #4246 

_What was the problem?_
It was unclear when the logs table was awaiting search results after a search filter was added. Table rows might appear to be stale.
_What was the solution?_
Add a search status that is updated when a search is performed and render a corresponding loading message for no results, loading, or loaded table data.

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [ ] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/logs/constants/index.ts [hunk 1]
```diff
@@ -204,3 +204,11 @@ export const HISTOGRAM_CENTRAL_REGION = 3 / 5
 export const HISTOGRAM_SHIFT = 1 / 5
 
 export const SECONDS_TO_MS = 1000
+
+export const LoadingMessages = [
```
Comment (AlirieGray) [inline ui/src/logs/constants/index.ts]: 😂

File: ui/src/logs/utils/table.ts [hunk 2]
```diff
@@ -157,3 +157,14 @@ export const applyChangesToTableData = (
     values: updatedValues,
   }
 }
+
+export const isEmptyInfiniteData = (data: {
+  forward: TableData
+  backward: TableData
+}) => {
+  return isEmptyTableData(data.forward) && isEmptyTableData(data.backward)
```
Comment (AlirieGray) [inline ui/src/logs/utils/table.ts]: a little confused by this naming, does this replace the `isTableEmpty()` from above?

File: ui/src/logs/utils/table.ts [hunk 3]
```diff
@@ -157,3 +157,14 @@ export const applyChangesToTableData = (
     values: updatedValues,
   }
 }
+
+export const isEmptyInfiniteData = (data: {
```
Comment (AlirieGray) [inline ui/src/logs/utils/table.ts]: a little confused by this naming (`isEmptyInfiniteData`). does this replace the isTableEmpty() from above?

File: ui/src/logs/constants/index.ts (hunk 1 above)
Comment (OfTheDelmer) [reply ui/src/logs/constants/index.ts]: haha

File: ui/src/logs/utils/table.ts (hunk 3 above)
Comment (OfTheDelmer) [reply ui/src/logs/utils/table.ts]: Yeah, it's now a util that helps check when to set searchStatus to `LoadingStatus.NoResults` when the tableInfiniteData is empty.

**Pull request merged**